### PR TITLE
UCT/IB/UD: UD mlx5 use res_domain

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**

### CODE REVIEW

File: src/uct/ib/ud/accel/ud_mlx5.c [hunk 1]
```diff
@@ -721,7 +721,7 @@ static UCS_CLASS_INIT_FUNC(uct_ud_mlx5_iface_t,
 
     ucs_trace_func("");
 
-    init_attr.res_domain_key = UCT_IB_IFACE_NULL_RES_DOMAIN_KEY;
+    init_attr.res_domain_key = UCT_IB_MLX5_RES_DOMAIN_KEY;
```
Comment (hoopoepg) [inline src/uct/ib/ud/accel/ud_mlx5.c]: this cause to use different keys for ud_mlx5/ib/cm... is it ok?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9028/ for details (Mellanox internal link).

File: src/uct/ib/ud/accel/ud_mlx5.c (hunk 1 above)
Comment (yosefe) [reply src/uct/ib/ud/accel/ud_mlx5.c]: verbs should have NULL key since it does not use res domain
and all mlx5 transports should have same non-NULL key

**Comment by brminich** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9043/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9046/ for details (Mellanox internal link).

**Pull request merged**